src/libs/ajax.js

```diff
@@ -51,11 +52,28 @@ const withCancellation = wrappedFetch => async (...args) => {
   }
 }
 
+//const captureRetryFailure = (...args) => _.once(() => {
+//      console.log('capture failure')
+//      const errorAddress = _.flow(
+//        _.values,
+//        _.find(v => _.includes(v, args[0]) && v)
+//      )([..._.values(getConfig()), 'https://storage.googleapis.com/'])
+//      console.log("errorAddress", errorAddress)
+//      Ajax().Metrics.captureEvent(Events.requestFailed, { test: errorAddress })
+//  })
+
 const withRetryOnError = _.curry(wrappedFetch => async (...args) => {
   const timeout = 5000
   const somePointInTheFuture = Date.now() + timeout
   const maxDelayIncrement = 1500
   const minDelay = 500
+  const captureFailure = _.once(() => {
+    const errorAddress = _.flow(
+      _.values,
+      _.find(v => _.includes(v, args[0]) && v)
+    )([..._.values(getConfig()), 'https://storage.googleapis.com/'])
```

Comment:
It looks like this is getting the API root for the URL of the request that failed? Why not capture the full URL to identify which endpoint failed?

Reply:
the agreed upon Idea was a need to make the metrics easer to group to gather, so focus on a shared root rather than each one having a unique url.